test/evp_test.c

```diff
@@ -3035,13 +3036,29 @@ static int run_file_tests(int i)
 
 OPT_TEST_DECLARE_USAGE("file...\n")
 
+static OSSL_PROVIDER *defltprov = NULL;
+static OSSL_PROVIDER *legacyprov = NULL;
+
 int setup_tests(void)
 {
     size_t n = test_get_argument_count();
 
     if (n == 0)
         return 0;
 
+    defltprov = OSSL_PROVIDER_load(NULL, "default");
+    if (!TEST_ptr(defltprov))
+        return 0;
+    legacyprov = OSSL_PROVIDER_load(NULL, "legacy");
```

Comment:
Needs to be guarded to deal with "no-legacy". There is no `OPENSSL_NO_LEGACY` but see:

https://github.com/openssl/openssl/blob/07822c515574223c2958fcddd52191cfbf9aefcd/test/build.info#L189-L191
https://github.com/openssl/openssl/blob/07822c515574223c2958fcddd52191cfbf9aefcd/test/evp_extra_test.c#L1256-L1260

Reply:
Its a bit more complicated than this. I would need to have some way of determining that algorithms are legacy since this test processes .txt files. (either with a extra field, or by seperating the legacy tests into their own .txt file)

Reply:
Added a Legacy=1 to the test data

Reply:
Also noticed mdc2 was broken with -no-legacy so addressed this also..